scripts/paper/generate_stwo_primitive_lookup_vs_naive_figure.py

```
@@ -0,0 +1,264 @@
+    parser.add_argument(
+        "--output-svg",
+        type=Path,
+        default=OUTDIR / "stwo-primitive-lookup-vs-naive-2026-04.svg",
+    )
+    parser.add_argument("--output-png", type=Path, default=None)
+    parser.add_argument("--output-pdf", type=Path, default=None)
+    args = parser.parse_args()
+
+    rows = read_rows(args.input_tsv)
+    if not rows:
+        raise SystemExit(f"no rows found in {args.input_tsv}")
```

Ruff (0.15.10)

[warning] Avoid specifying long messages outside the exception class

(TRY003)